### PR TITLE
Bugfix/dapp connect token screen shows transfer header

### LINKED ISSUE #2675: dApp: Transfer screen header is being displayed on select token screen
## Todo
Create a separate route specifically for this entry point to the dApp.

## Steps to Reproduce
1. Connect to the dApp with no previously connected tokens

## Expected Result
- Select token header

## Actual Result
- Transfer header

## Additional Infos (Optional) 
Screenshot 2021-04-19 at 11 32 33

### PR DESCRIPTION
**Pull request by weilbith**
**Thank you for submitting this pull request :)**

Fixes #2675

**Short description**

Solution description with background details in the linked issue. All other infos in the according commits. 
This includes some nice improvements of the router guards.

**Definition of Done**

- [x] Steps to manually test the change have been documented
- [x] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1. Connect to the dApp and use an account you never used for it before
2. Make sure you see the route to connect your frist token
3. Try to access the transfer route for example and make sure you get redirected again
4. Make sure you can still access all account routes (e.g. to transfer ETH or deposit to UDC)
5. Add your first token/open a channel and make sure all routes are accessible for that
6. Make sure you finally can access the transfer route (also after reload/re-connect)

### CODE REVIEW

**Comment by weilbith** [comment]
I'll rebase on master to get the end-to-end fix of #2738 and also resolve the merge conflict.

File: raiden-dapp/src/router/guards/connection.ts [hunk 1]
```diff
@@ -1,26 +1,23 @@
 import { RouteNames } from '../route-names';
 import type { NavigationGuardChild } from './types';
 
-const accountRoutes = [
+const freeAccountRoutes = [
```
Comment (taleldayekh) [inline raiden-dapp/src/router/guards/connection.ts:4]: `unprotectedAccountRoutes`?

**Review by taleldayekh — APPROVED** [review]
Thank you for fixing this 🕷️

File: raiden-dapp/tests/unit/router/guards/no-connected-token.spec.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  accountSettingsRoute,
+  accountWithdrawalRoute,
+  accountUdcRoute,
+];
+
+const connectTokenRoutes = [selectTokenRoute, selectHubRoute, openChannelRoute];
+
+const tokens = {
+  '0xtoken': {
+    '0xpartner': 1, // just something...
```
[the remark this answers is not in the document]
Comment (weilbith) [reply raiden-dapp/tests/unit/router/guards/no-connected-token.spec.ts:40]: Very good point.

File: raiden-dapp/tests/unit/router/guards/no-connected-token.spec.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import { createLocalVue } from '@vue/test-utils';
+import Vuex, { Store } from 'vuex';
+
+import { redirectIfNoConnectedToken } from '@/router/guards/no-connected-token';
+import { RouteNames } from '@/router/route-names';
+import type { CombinedStoreState } from '@/store';
+
+import { transformRouteConfigsToRoutes } from '../../utils/router-utils';
+
+const localVue = createLocalVue();
```
[the remark this answers is not in the document]
Comment (weilbith) [reply raiden-dapp/tests/unit/router/guards/no-connected-token.spec.ts:10]: Yes, you can only create a store (`new Vuex.Store()`) when you have registered Vuex to Vue. As we don't have a component here this is the preferable approach. In general this is what the Vue dev-docs recommend to do. I would like to use it way more in our unit tests. But its not straight forward everywhere and will be a lot of work to make it work.

If that was helpful enough, plea close the conversation. :upside_down_face:

File: raiden-dapp/src/router/guards/connection.ts (hunk 1 above)
Comment (weilbith) [reply raiden-dapp/src/router/guards/connection.ts:4]: I was so unhappy with the naming. Thank you very much for the suggestion. 🙏🏾

Comment (weilbith) [reply raiden-dapp/src/router/guards/connection.ts]: ✔️

File: raiden-dapp/tests/unit/router/guards/no-connected-token.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (weilbith) [reply raiden-dapp/tests/unit/router/guards/no-connected-token.spec.ts]: ✔️

**Comment by weilbith** [comment]
This PR now includes the bugfix for the failing end-to-end tests. I'm pretty happy with the solution. 🙃 
Once this PR is done and merged, I'll rebase the other open PRs on top of it.

**Comment by weilbith** [comment]
I ran the CI pipe twice in a row and both runs succeed. Looks good.

**Review by taleldayekh — APPROVED** [review]
Nice Thore 👍🏽

**Pull request merged**